mpmp/src/model/Player.java

```diff
@@ -51,8 +63,13 @@ public String toString() {
 	 * Adds or removes money from a player. If the player can't pay, false is returned.
 	 */
 	public boolean addMoney(int sum) {
-		if(money + sum >= 0) {
-			money += sum;
+		if(cash + hyp + sum >= 0) {
+			cash += sum;
+
+			/* cash exhausted; change to hypothecs instead */
+			if(cash < 0)
+				hyp = -cash;
```

Comment:
I kjnow it is a little convoluted; I think we need to place a comment here. It is really like you describe: `hyp` is the money gained from hypothecs, `cash` is your normal money. The total money is the sum of `cash` and `hyp` (as can be seen in the code). The `if` checks whether we have a positive amount of money after adding a certain `sum`. Note that `sum` is often negative.